src/gradualizer_db.erl

```diff
@@ -265,6 +278,32 @@ import_prelude(State = #state{loaded = Loaded}) ->
     %% are loaded on demand
     State1#state{loaded = Loaded}.
 
+-spec import_extra_specs(filelib:filename(), state()) -> state().
+import_extra_specs(Dir, State = #state{loaded = Loaded}) ->
+    FormsByModule = get_module_forms_tuples(Dir),
+    %% Import forms each of the modules to override
+    State1 = lists:foldl(fun ({Module, Forms}, StateAcc) ->
+                                 import_absform(Module, Forms, StateAcc)
+                         end,
+                         State,
+                         FormsByModule),
+    %% Mark the just overridden modules as not yet loaded, to make sure they
+    %% are loaded on demand
+    State1#state{loaded = Loaded}.
+
+-spec get_module_forms_tuples(filelib:filename()) -> [{module(), gradualizer_file_utils:abstract_forms()}].
+get_module_forms_tuples(Dir) ->
+    Files = filelib:wildcard(filename:join([Dir, "*.specs.erl"])),
+    lists:map(fun get_module_and_forms/1, Files).
+
+%% Parses and returns the forms of a file along with the module given in the -module attribute
+-spec get_module_and_forms(file:filename()) -> {module(), gradualizer_file_utils:abstract_forms()}.
+get_module_and_forms(File) ->
+    {ok, Forms} = gradualizer_file_utils:get_forms_from_erl(File, []),
+    [{attribute, _, file, _},
+     {attribute, _, module, Module} | _] = Forms,
+    {Module, Forms}.
```

Comment:
Can these functions be exported in `gradualizer_prelude_parse_trans` and used here? To avoid duplicated code while not adding any dependency to the parse transform. They are copied from there, right?